### PR TITLE
Several small SFX tweaks

### PR DESCRIPTION
**Pull request by LittleNyanCat**
## About the PR
The following sounds now have pitch variations:
- Computer keyboard sound when opening the GUI.
- Eating and drinking sounds.
- Vending machine vend sound.
- Clown shoes, cowboy shoes and their variants.
- Honkbots and the H.O.N.K mech

The vending machine vend sound effect has been replaced by one from TG.

Adding water/reagents to a hydro basin with a spillable container now plays a sloshing sound.

## Why
- Pitch variations are a small touch that makes sounds a little less boring, especially when played repeatedly (eating, drinking).
- The current vend sound is barely audible and the TG one is more satisfying.
- The slosh sound adds to the immersion.
- The clown shoes always sounded like something was missing and this was it.

## Media

https://github.com/space-wizards/space-station-14/assets/29161338/74971215-d2ab-498f-858b-2207bc7a7d3b

https://github.com/space-wizards/space-station-14/assets/29161338/78d6687a-dd59-447f-ad8f-c034cc940d48

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

no changelog :(

### CODE REVIEW

**Comment by LittleNyanCat** [comment]
Why the fuck is it failing with something relating to atmos I didn't touch it at all

**Comment by Tayrtahn** [comment]
Congrats! You found one of our heisentests. They're caused by errors that only show up in very specific conditions, usually with some form of randomness behind them.

(I poked the test runner to try again)

**Comment by LittleNyanCat** [comment]
Oh golly, how wonderful!

Well at least it passed now.

File: Content.Server/Botany/Systems/PlantHolderSystem.cs [hunk 1]
```diff
@@ -261,6 +262,15 @@ private void OnInteractUsing(Entity<PlantHolderComponent> entity, ref InteractUs
...
         }
 
+        //Only play splosh sound when using containers that would be "dumped" into the basin (as opposed to ie. sprayed on)
+        if (TryComp(args.Used, out SpillableComponent? spill) && _solutionContainerSystem.TryGetRefillableSolution(args.Used, out var solCpmp, out var sol))
+        {
+            if(component.WateringSound != null && sol.Volume > 0)
+            {
+                _audio.PlayPvs(component.WateringSound, uid);
+            }
+        }
```
Comment (mirrorcult) [inline Content.Server/Botany/Systems/PlantHolderSystem.cs:272]: this feels like a weird place for this logic ? none of the plantholder stuff actually handles the refilling, so this is basically just hoping that the code path of this megafunction eventually hits this and plays the sound

would make more sense imo for it to be on the refillable itself (or whatever handles this behavior i dont remember)

**Review by mirrorcult — DISMISSED** [review]

File: Content.Server/Botany/Systems/PlantHolderSystem.cs (hunk 1 above)
Comment (LittleNyanCat) [reply Content.Server/Botany/Systems/PlantHolderSystem.cs:272]: I completely agree with that, I looked around the solution container code and asked on the Discord if there was any event for that, but it doesn't seem to exist for now. I'd like to implement that myself at some point but I'm just not well versed with the codebase & engine enough for that yet.
 
The code that handles spraying (directly above mine) also plays their sound here (and also handles the reagent transfer, which should probably happen elsewhere too), so I reckoned it should be fine until someone can come in and code it proper

This feels kinda hacky but seems to work, my main worry was the solution in the container being emptied before my code ran, but it hasn't been an issue in my testing.

Comment (LittleNyanCat) [reply Content.Server/Botany/Systems/PlantHolderSystem.cs:272]: Nevermind, looking harder I think I found that the event I was looking for does indeed exist, gonna work on it now.

Comment (LittleNyanCat) [reply Content.Server/Botany/Systems/PlantHolderSystem.cs]: There, just pushed code that properly uses events instead of that hacky solution

Comment (LittleNyanCat) [reply Content.Server/Botany/Systems/PlantHolderSystem.cs]: The event I was using earlier was the solution transfer attempt, which is non ideal and might cause issues in the future.

It turns out that implementing an event for when the solution is successfully transferred wasn't nearly as hard as I thought it was gonna be, so now things are properly implemented as they should've been from the start.

**Comment by ps3moira** [comment]
love this

File: Content.Server/Botany/Components/PlantHolderComponent.cs [hunk 2]
```diff
@@ -23,6 +24,9 @@ public sealed partial class PlantHolderComponent : Component
     [DataField("lastCycle", customTypeSerializer: typeof(TimeOffsetSerializer))]
     public TimeSpan LastCycle = TimeSpan.Zero;
 
+    [DataField("wateringSound")]
```
Comment (EmoGarbage404) [inline Content.Server/Botany/Components/PlantHolderComponent.cs:27]:
```suggestion
    [DataField]
```
the datafield string is just inferred from the variable name

File: Content.Server/Botany/Systems/PlantHolderSystem.cs [hunk 3]
```diff
@@ -356,6 +360,19 @@ private void OnInteractUsing(Entity<PlantHolderComponent> entity, ref InteractUs
...
+    {
+        var (uid, component) = entity;
+
+        //Only play splosh sound when using containers that would be dumped into the basin (as opposed to ie. sprayed on)
+        if (TryComp(args.From, out SpillableComponent? spill))
+        {
+            if(component.WateringSound != null)
+            {
+                _audio.PlayPvs(component.WateringSound, uid);
+            }
```
Comment (EmoGarbage404) [inline Content.Server/Botany/Systems/PlantHolderSystem.cs:373]:
```suggestion
            _audio.PlayPvs(component.WateringSound, uid);
```
you dont need to check for null, as all the audio events take in nullable soundspecifiers.

**Review by EmoGarbage404 — DISMISSED** [review]

**Comment by LittleNyanCat** [comment]
I can't believe it took me this long to realize that the clown's shoes didn't pitch variations and that's why they sounded so off in SS14... Fixed!

**Review by AJCM-git — APPROVED** [review]

**Comment by metalgearsloth** [comment]
Merge after https://github.com/space-wizards/space-station-14/pull/27481

File: Content.Server/Botany/Systems/PlantHolderSystem.cs [hunk 4]
```diff
@@ -55,6 +57,7 @@ public override void Initialize()
         SubscribeLocalEvent<PlantHolderComponent, ExaminedEvent>(OnExamine);
         SubscribeLocalEvent<PlantHolderComponent, InteractUsingEvent>(OnInteractUsing);
         SubscribeLocalEvent<PlantHolderComponent, InteractHandEvent>(OnInteractHand);
+        SubscribeLocalEvent<PlantHolderComponent, SolutionTransferSuccessEvent>(OnSolutionAdded);
```
Comment (metalgearsloth) [inline Content.Server/Botany/Systems/PlantHolderSystem.cs:60]:
```suggestion
        SubscribeLocalEvent<PlantHolderComponent, SolutionTransferredEvent>(OnSolutionAdded);
```

File: Content.Server/Botany/Systems/PlantHolderSystem.cs [hunk 5]
```diff
@@ -356,6 +360,16 @@ private void OnInteractUsing(Entity<PlantHolderComponent> entity, ref InteractUs
         }
     }
 
+    private void OnSolutionAdded(Entity<PlantHolderComponent> entity, ref SolutionTransferSuccessEvent args)
```
Comment (metalgearsloth) [inline Content.Server/Botany/Systems/PlantHolderSystem.cs:363]:
```suggestion

    private void OnSolutionAdded(Entity<PlantHolderComponent> entity, ref SolutionTransferredEvent args)
```

**Comment by LittleNyanCat** [comment]
tfw I thought this would be a quick pr that I would be done with in two weeks

**Comment by slarticodefast** [comment]
Hey sorry for the long wait. This PR kinda got lost because it was waiting on another PR which got closed unexpectedly.
I fixed the merge conflict for you and did some minor cleanup:
- the slosh.ogg sound already existed in a different folder
- I removed the check for SpillableComponent on solution transfer since it was a bandaid bugfix. I will make a separate PR to fix the double interaction with the plant holder that causes two sounds to play.
- some code convention stuff

Otherwise this is a great addition, thank you for your contribution and your first PR!

@metalgearsloth this should be good to merge now

**Review by slarticodefast — APPROVED** [review]

**Comment by chromiumboy** [comment]
Thank you for your contribution, Sloth has given his approval for this PR

**Comment by LittleNyanCat** [comment]
Oh neat, this got merged!

I thought it was just cast to the bowels of PR ether for the end of eternity

**Comment by slarticodefast** [comment]
Yeah, sorry for the wait. It kinda got stuck in review hell because it was waiting on a refactor that was never finished.

**Pull request merged**